client/main.lua

```diff
@@ -1,176 +1,186 @@
+                description = Lang:t('menu.juice_items_needed', { amount = sharedConfig.grapesNeeded }),
+                icon = 'bottle-droplet',
+                onSelect = function()
+                    juiceProcessing()
+                end,
+            }
+        }
+    })
+
+    lib.showContext('processingMenu')
+end
 
-	if finishedWine then
-		lib.showTextUI(Lang:t("task.get_wine"), {position = 'right'})
-		if IsControlJustReleased(0, 38) and not LocalPlayer.state.invBusy then
-			TriggerServerEvent("qb-vineyard:server:receiveWine")
-			finishedWine = false
-			loadIngredients = false
-		    wineStarted = false
-		end
-		return
-	end
+lib.zones.box({
+    coords = config.locations.vineyardProcessing.coords,
+    size = vec3(1.6, 1.4, 3.2),
+    rotation = 346.25,
+    debug = config.debugPoly,
+    onExit = function()
+        lib.hideTextUI()
+    end,
+    onEnter = function ()
+        lib.showTextUI(Lang:t('task.vineyard_processing'))
+    end,
+    inside = function()
+        if IsControlJustReleased(0, 38) then
+            processingMenu()
+        end
+    end,
+})
 
-	lib.showTextUI(Lang:t("task.load_ingrediants"), {position = 'right'})
-	if IsControlJustReleased(0, 38) then
-		lib.callback('qb-vineyard:server:loadIngredients', false, function(result)
-			if result then loadIngredients = true end
-		end)
-	end
+for _, coords in pairs(config.grapeLocations) do
+    lib.zones.box({
+        coords = coords,
+        size = vec3(1, 1, 1),
+        rotation = 40,
+        debug = config.debugPoly,
+        onExit = function()
+            lib.hideTextUI()
+        end,
+        onEnter = function()
+            lib.showTextUI(Lang:t("task.start_task"))
+        end,
+        inside = toPickGrapes,
+    })
 end
 
-local function juiceWork()
-	if IsControlJustReleased(0, 38) then
-		lib.callback('qb-vineyard:server:grapeJuice', false, function(result)
-			if result then
-                prepareAnim()
-                grapeJuiceProcess()
-            end
-		end)
-		return false
-	end
-	return true
+local function init()
+    if config.useBlips then setLocationsBlip() end
 end
 
-lib.zones.poly({
-	points = config.locations.wine.zones,
-	thickness = 2,
-	debug = config.debugPoly,
-	onExit = exitZone,
-	onEnter = enterZone,
-	inside = workWine
-})
-
-lib.zones.poly({
-	points = config.locations.grapejuice.zones,
-	thickness = 2,
-	debug = config.debugPoly,
-	onExit = exitZone,
-	onEnter = enterZone,
-	inside = juiceWork
-})
+RegisterNetEvent('QBCore:Client:OnJobUpdate', function()
+    init()
+end)
 
-for _, coords in pairs(grapeLocations) do
-	lib.zones.box({
-		coords = coords,
-		size = vec3(1, 1, 1),
-		rotation = 40,
-		debug = config.debugPoly,
-		onExit = exitZone,
-		onEnter = enterZone,
-		inside = toPickGrapes,
-	})
-end
+AddEventHandler('onResourceStart', function(resourceName)
+    if GetCurrentResourceName() ~= resourceName then return end
+    if isLoggedIn then init() end
+end)
```

Comment:
```suggestion
if isLoggedIn then init() end
```
Can be simplified since this code runs when the resource starts anyway.